airflow/jobs/scheduler_job.py

```diff
@@ -1057,30 +1027,34 @@ def _find_executable_task_instances(self, simple_dag_bag, states, session=None):
         TI = models.TaskInstance
         DR = models.DagRun
         DM = models.DagModel
-        ti_query = (
-            session
-            .query(TI)
-            .filter(TI.dag_id.in_(simple_dag_bag.dag_ids))
+        ti_query = BAKED_QUERIES(
+            lambda session: session.query(TI).filter(
+                TI.dag_id.in_(simple_dag_bag.dag_ids)
+            )
             .outerjoin(
                 DR,
                 and_(DR.dag_id == TI.dag_id, DR.execution_date == TI.execution_date)
             )
-            .filter(or_(DR.run_id == None,  # noqa: E711 pylint: disable=singleton-comparison
-                    not_(DR.run_id.like(BackfillJob.ID_PREFIX + '%'))))
+            .filter(or_(DR.run_id.is_(None),
+                        not_(DR.run_id.like(BackfillJob.ID_PREFIX + '%'))))
```

Comment:
I really don't like filtering with the like expression. This makes the query very difficult to optimize. It is not possible to store it in a simple data structure. We have to have a very complex binary tree, but which takes more memory than a simple structure with 3 values. Which causes other problems, e.g. unbalanced tree, and thus performance degradation.

Reply:
I agree, and we should store a "run_type" column or something similar on the dag run table.

But this is unchanged behaviour.

Reply:
I created ticket: 
https://issues.apache.org/jira/browse/AIRFLOW-6242
A small step forward.